lib/blockscore/base.rb

```diff
@@ -0,0 +1,60 @@
+require 'active_support/core_ext/class'
+require 'active_support/core_ext/string/inflections'
+require 'blockscore/connection'
+
+module BlockScore
+  class Base
+    extend BlockScore::Connection
+    
+    class_attribute :api_key, :resource, :version
+
+    def self.inherited(base)       
+      base.resource = base.to_s.split('::').last.underscore
+    end
+
+    def initialize(options = {})
+      options[:class] = resource
+      @attrs = options
+    end
+
+    def refresh
+      r = self.class.retrieve(id)
+      if r != BlockScoreError
+        instance_variable_set(:@attrs, r.instance_variable_get(:@attrs))
+
+        true
+      else
+        false
+      end
+    end
+
+    def self.auth(api_key)
+      api_key = api_key
+      BlockScore::Connection.api_key = api_key
+    end
+    
+    def self.api_url
+      'https://api.blockscore.com/'
+    end
+
+    def self.endpoint
+      if self == BlockScore::Base
+        raise NotImplementedError.new('Base is an abstract class. You should perform actions on its subclasses (Candidate, Company, Person, etc.)')
```

Comment:
Metrics/LineLength: Line is too long. [147/100]<br>Style/SignalException: Use `fail` instead of `raise` to signal exceptions.